### DISCUSSION TITLE
Point operations in float mode

### DISCUSSION THREAD

**Discussion by kamocat** [python-pillow/Pillow#7550] [General]
The documentation for `Image.point()` says you can pass it functions or lookup tables.
The examples show a simple linear lambda function using multiply and add operators.
However, if I include a divide operator, it all breaks down. For example, `im.point(lambda x: 500 / x)` results in
`TypeError: unsupported operand type(s) for /: 'int' and '_E'`

(My actual formula is somewhat more complex, involving several operators and a logarithm to convert radiated IR into measured temperature)

On the other hand, lookup tables are not supported for float type images, as evidenced by the [source code](https://pillow.readthedocs.io/en/stable/_modules/PIL/Image.html#Image.point), and I do not wish to lose accuracy by casting to integers.

Happily, I found a concise way to do this:
`im.putdata(np.vectorize(my_function)(im.getdata()))`

I was intending to post this as a question, but I found my answer as I was providing evidence of the problem. Hopefully this will help someone else who is struggling to use Image.point() in the future.
As a side note, I never found a clear example of what format the lookup table should be. Is it just a list of values, addressed by index?

**Comment by radarhere** [comment]
> I never found a clear example of what format the lookup table should be. Is it just a list of values, addressed by index?

It's just a list of values, yes.

https://github.com/python-pillow/Pillow/blob/12022fe54e0ab4169629b091305f2e0195f84895/Tests/test_image_point.py#L11
https://github.com/python-pillow/Pillow/blob/12022fe54e0ab4169629b091305f2e0195f84895/Tests/test_image_point.py#L44-L45
